src/Microsoft.Azure.WebJobs.Host/Indexers/FunctionIndexProvider.cs

```diff
@@ -2,27 +2,18 @@
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
 using System;
-using System.Collections.Generic;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.Azure.WebJobs.Host.Bindings;
 using Microsoft.Azure.WebJobs.Host.Executors;
-using Microsoft.Azure.WebJobs.Host.Storage;
+using Microsoft.Azure.WebJobs.Host.Timers;
 using Microsoft.Azure.WebJobs.Host.Triggers;
 
 namespace Microsoft.Azure.WebJobs.Host.Indexers
 {
     internal class FunctionIndexProvider : IFunctionIndexProvider
     {
-        private readonly ITypeLocator _typeLocator;
-        private readonly ITriggerBindingProvider _triggerBindingProvider;
-        private readonly IBindingProvider _bindingProvider;
-        private readonly IJobActivator _activator;
-        private readonly IFunctionExecutor _executor;
-        private readonly IExtensionRegistry _extensions;
-        private readonly SingletonManager _singletonManager;
-        private readonly TraceWriter _trace;
-
+        private Func<CancellationToken, Task<IFunctionIndex>> _initializeIndex;
```

Comment:
was routing parameters through this class, saw that it was only serving as an intialization wrapper, encapsulated in a lambda